optuna_dashboard/_preferential_history.py

```diff
@@ -78,3 +79,22 @@ def report_history(
         key=key,
         value=json.dumps(history),
     )
+    return history_id
+
+
+def switching_history(study_id: int, storage: BaseStorage, uuid: str, enable: bool) -> None:
+    system_attrs = storage.get_study_system_attrs(study_id)
+    history: History = json.loads(system_attrs.get(_SYSTEM_ATTR_PREFIX_HISTORY + uuid, ""))
+    if enable:
+        preferences = [
+            (best, history["clicked"])
+            for best in history["candidates"]
+            if best != history["clicked"]
+        ]
+        storage.set_study_system_attr(
+            study_id, _SYSTEM_ATTR_PREFIX_PREFERENCE + history["preference_id"], preferences
+        )
+    else:  # disable
+        storage.set_study_system_attr(
+            study_id, _SYSTEM_ATTR_PREFIX_PREFERENCE + history["preference_id"], []
```

Comment:
The function name `switching_history` sounds a bit ambiguous to me. How about splitting this function into following to functions?

* `remove_history(storage, study_id, history_id)` which means `switching_history(..., enable=False)`.
* `restore_history(storage, study_id, history_id)` which means `switching_history(..., enable=True)`.

Reply:
I think we should split the api endpoint as well.